### PR TITLE
#3649 Connector actions (start, stop, delete, configure)

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3649 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/common/Definitions/RenderDefinition.js [hunk 1]
```diff
@@ -54,22 +65,22 @@ const RenderDefinition = props => {
...
 
     switch (permission) {
-      case READ_ONLY:
+      case 'READ_ONLY':
         disabled = true;
         break;
 
-      case CREATE_ONLY:
+      case 'CREATE_ONLY':
         disabled = defType === CREATE_ONLY ? false : true;
```
Comment (eechih) [inline ohara-manager/client/src/components/common/Definitions/RenderDefinition.js]: should be written like `disabled = defType !== CREATE_ONLY`

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js [hunk 2]
```diff
@@ -158,6 +159,12 @@ const SettingDialog = props => {
   const { title = '', classInfo = {} } = data;
   const [expanded, setExpanded] = useState(null);
   const [selected, setSelected] = useState(null);
+  const { data: currentConnectors } = useConnectorState();
+  const { data: currentTopics } = useTopicState();
+  const { updateConnector } = useConnectorActions();
+  const tagerConnector = currentConnectors.filter(
+    connector => connector.className === classInfo.className,
+  )[0];
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/SettingDialog.js]: should use find() instead

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js [hunk 3]
```diff
@@ -184,13 +191,37 @@ const SettingDialog = props => {
...
+      group: tagerConnector.group,
+      ...values,
+    });
+    handleClose();
   };
 
   const { RenderForm, formHandleSubmit, refs } = RenderDefinitions({
+    topics: currentTopics,
     Definitions: groups.sort(),
+    initialValues: tagerConnector,
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/SettingDialog.js]: is typo?

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js [hunk 4]
```diff
@@ -184,13 +191,37 @@ const SettingDialog = props => {
...
 
-  const onSubmit = async (values, form) => {
-    form.reset();
-    return values;
+  const onSubmit = async values => {
+    if (values.topicKeys) {
+      if (values.topicKeys.startsWith('T')) {
+        const privateTopic = currentTopics.filter(
+          topic => topic.tags.displayName === values.topicKeys,
+        )[0];
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/SettingDialog.js]: 不太確定這一段邏輯的目的，請問 values.topicKeys 是 array 還是 string?

Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/SettingDialog.js]: 原本是array 在經過final form 會只剩string 但post需要object keys.

File: ohara-manager/client/src/components/common/Definitions/RenderDefinition.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/common/Definitions/RenderDefinition.js]: thanks

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/SettingDialog.js]: yes ~~
thanks

File: ohara-manager/client/src/components/Pipeline/SettingDialog.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/SettingDialog.js]: ok thanks

**Pull request merged**